##### 3.7 Access the horseshoe crab data in Table 3.2 at www.stat.ufl.edu/∼aa/intro-cda/appendix.html. Let Y = 1 if a crab has at least one satellite, and let Y = 0 otherwise. Using weight as the predictor, fit the linear probability model.(Weight is in grams.)

$\pi$ = -0.1449 + 0.0003


In [2]:
library(dplyr)
HorseshoeCrab<-read.csv("HorseshoeCrab.csv")
head(HorseshoeCrab)
HorseshoeCrab$satellites<- ifelse(HorseshoeCrab$satellites>=1, 1,0) #making satellites binary
linear_prob_model<-summary(glm(satellites~weight,family=gaussian,data=HorseshoeCrab))
round(linear_prob_model$coeff,4)

color spine width satellites weight Y
1 3     3     28.3  8          3050   1
2 4     3     22.5  0          1550   0
3 2     1     26.0  9          2300   1
4 4     3     24.8  0          2100   0
5 4     3     26.0  4          2600   1
6 3     3     23.8  0          2100   0

Estimate Std. Error t value Pr(>|t|)
(Intercept) -0.1449  0.1472     -0.9845 0.3263  
weight       0.0003  0.0001      5.4916 0.0000

###### a. Use ordinary least squares. Interpret the parameter estimates. Find the predicted probability at the highest observed weight of 5.20 kg.Comment.


In [3]:
#1 gram = .001kg
#5200 grams = 5.2kg

ols.horseshoecrab<-lm(satellites ~ weight, data=HorseshoeCrab) #lm() for least squares while glm() for maximum likelihood estimate 
round(ols.horseshoecrab$coeff,4)

predict(ols.horseshoecrab,data.frame(weight = 5200))


(Intercept)      weight 
    -0.1449      0.0003

1 
1.533186

${\hat{\beta}}$=.0003

For each 1 gram , there is a .0003 increase in predicted probability that a crab has at least one satellite.

$\alpha$= -0.1449

When the weight of the  female horse shoe crab is 0 g, the predicted probability that a crab has at least one satellite is -0.1449 which is not possible 
because that number is outside of the [0,1] probability range.

The predicted probability at the highest observed weight of 5.2kh (5200g) is 1.533 which exceeds the [0,1] probability range.


###### b. Attempt to fit the model using ML, treating Y as binomial. What does your software report? [The failure is due to a fitted probability falling outside the (0, 1) range.]

In [4]:
ml_model<-summary(glm(satellites~weight,family=gaussian,data=HorseshoeCrab))
round(ml_model$coeff,4)

Estimate Std. Error t value Pr(>|t|)
(Intercept) -0.1449  0.1472     -0.9845 0.3263  
weight       0.0003  0.0001      5.4916 0.0000

$\pi$ = -0.1449 + 0.0003
But like 7b we can see the predicted probability at the highest observed weight of 5.2kh (5200g) is 1.533 which exceeds the [0,1] probability range.

###### c. Fit the logistic regression model. Show that the estimated logit at a weight of 5.20 kg equals 5.74. Show that πˆ = 0.9968 at that point by checking that log[ ˆπ/(1 − ˆπ )] = 5.74 when πˆ = 0.9968.

Logistic regression model: logit[ ˆπ (x)]=−3.69 + 0.0018x

We can see from the code below that log[ ˆπ/(1 − ˆπ )] = 5.74 when πˆ = 0.9968.

In [5]:
#1 gram = .001kg
#5200 grams = 5.2kg

colnames(HorseshoeCrab)
head(HorseshoeCrab)
library(dplyr)
HorseshoeCrab<-read.csv("HorseshoeCrab.csv")
HorseshoeCrab$satellites<- ifelse(HorseshoeCrab$satellites>=1, 1,0) #making satellites binary

LRM_HorseshoeCrab <- glm(satellites~weight,family=binomial,data=HorseshoeCrab)
LRM_HorseshoeCrab$coef
predict(LRM_HorseshoeCrab,data.frame(weight = 5200)) #5200 grams = 5.2 kg

#Showing that  πˆ = 0.9968
log((0.9968)/(1-0.9968))

[1] "color"      "spine"      "width"      "satellites" "weight"    
[6] "Y"

color spine width satellites weight Y
1 3     3     28.3  1          3050   1
2 4     3     22.5  0          1550   0
3 2     1     26.0  1          2300   1
4 4     3     24.8  0          2100   0
5 4     3     26.0  1          2600   1
6 3     3     23.8  0          2100   0

(Intercept)       weight 
-3.694726392  0.001815145

1 
5.744025

[1] 5.741399

#### 3.8 Refer to the previous exercise for the horseshoe crab data.

##### a. Report the fit for the probit model, with weight predictor.

Let Y = 1 if a crab has at least one satellite, and let Y = 0 otherwise.

In [6]:
HorseshoeCrab<-read.csv("HorseshoeCrab.csv")
colnames(HorseshoeCrab)
HorseshoeCrab$satellites<- ifelse(HorseshoeCrab$satellites>=1, 1,0) #assigning satellites 0s and 1s 

#probit model
probit_model_horseshoecrab<-glm(satellites~weight,family=binomial(link="probit"),HorseshoeCrab);probit_model_horseshoecrab

[1] "color"      "spine"      "width"      "satellites" "weight"    
[6] "Y"


Call:  glm(formula = satellites ~ weight, family = binomial(link = "probit"), 
    data = HorseshoeCrab)

Coefficients:
(Intercept)       weight  
  -2.238295     0.001099  

Degrees of Freedom: 172 Total (i.e. Null);  171 Residual
Null Deviance:	    225.8 
Residual Deviance: 195.5 	AIC: 199.5

probit[$\hat{\pi}$(x)]=-2.238+0.001099(weight)

##### b. Find πˆ (the fitted probability) at the highest observed weight, 5.20 kg.

In [7]:
#5.2 g = 5200g 
  
predict(probit_model_horseshoecrab,data.frame(weight=5200)) #probit value 
pnorm(predict(probit_model_horseshoecrab,data.frame(weight=5200))) #left tail probability

1 
3.476709

1 
0.9997462

$\hat{\pi}$(x=5.20) ~ 0.013
At observed weight x = 5.20, the probit equals approximately 3.47 which corresponds to a fitted probability of approximately 1. 

##### c. Describe the weight effect by finding the difference between the πˆ values at the upper and lower quartiles of weight, 2.85 and 2.00 kg.

In [9]:
predict(probit_model_horseshoecrab,data.frame(weight=c(2.85*1000,2.00*1000))) #probit value of weights 2.85kg (2850g) and 2.00 kg (2000g)
pnorm(predict(probit_model_horseshoecrab,data.frame(weight=2.85*1000)))-pnorm(predict(probit_model_horseshoecrab,data.frame(weight=2.00*1000)))

1           2 
 0.89396684 -0.04021637

1 
0.3303699

The weight effect is 0.33 - the difference between the probabality of quartiles of weight 2.85 and 2.00 kg.

##### d. Interpret the parameter estimates using characteristics of the normal cdf that describes the response curve.

weight: 0.001099
(Intercept): -2.238

This probit fit model corresponds to a normal cdf having mean (− ˆα/βˆ) = 2036.4 and standard deviation 1/|βˆ| = 909.92